src/hooks/student-list-section/enroll-student-manually/enroll-student.tsx

```diff
@@ -0,0 +1,99 @@
+//     root_user: z.boolean()
+//   }),
+//   student_extra_details: z.object({
+//     fathers_name: z.string(),
+//     mothers_name: z.string(),
+//     parents_mobile_number: z.string(),
+//     parents_email: z.string(),
+//     linked_institute_name: z.string()
+//   }),
+//   institute_student_details: z.object({
+//     institute_id: z.string(),
+//     package_session_id: z.string(),
+//     enrollment_id: z.string(),
+//     enrollment_status: z.string(),
+//     enrollment_date: z.string(),
+//   })
+// });
+
+interface EnrollStudentParams {
+    formData: {
+        stepOneData: StepOneData | null;
+        stepTwoData: StepTwoData | null;
+        stepThreeData: StepThreeData | null;
+        stepFourData: StepFourData | null;
+        stepFiveData: StepFiveData | null;
+    };
+    packageSessionId: string;
+}
+
+export const useEnrollStudent = () => {
+    return useMutation({
+        mutationFn: async ({ formData, packageSessionId }: EnrollStudentParams) => {
+            const requestBody = {
+                user_details: {
+                    username: formData.stepFiveData?.username || "",
+                    email: formData.stepThreeData?.email || "",
+                    full_name: formData.stepTwoData?.fullName || "",
+                    address_line: "N/A",
+                    city: formData.stepThreeData?.city || "",
+                    mobile_number: formData.stepThreeData?.mobileNumber || "",
+                    // date_of_birth: formData.stepTwoData.dateOfBirth || "",
+                    date_of_birth: "",
+                    gender: formData.stepTwoData?.gender || "",
+                    password: formData.stepFiveData?.password || "",
+                    profile_pic_file_id: formData.stepOneData?.profilePicture?.id || "",
+                    roles: ["STUDENT"],
+                    root_user: false,
+                },
+                student_extra_details: {
+                    fathers_name: formData.stepFourData?.fatherName || "",
+                    mothers_name: formData.stepFourData?.motherName || "",
+                    parents_mobile_number: formData.stepFourData?.guardianMobileNumber || "",
+                    parents_email: formData.stepFourData?.guardianEmail || "",
+                    linked_institute_name: formData.stepTwoData?.collegeName || "",
+                },
+                institute_student_details: {
+                    institute_id: INSTITUTE_ID,
+                    package_session_id: packageSessionId,
+                    enrollment_id: formData.stepTwoData?.enrollmentNumber || "",
+                    enrollment_status: "ACTIVE",
+                    enrollment_date: new Date().toISOString(),
+                },
+            };
+
+            const response = await authenticatedAxiosInstance.post(
+                ENROLL_STUDENT_MANUALLY,
+                requestBody,
+            );
+
+            return response.data;
+        },
+    });

```

Comment:
This might not work since you are returning useMutation function directly. Change this code and write API logic in a seperate file. Use suspense query of tanstack since mutation is not used for getting the data.